src/index.tsx

```diff
@@ -127,15 +121,23 @@ else {
                 return []
             }      
         }
-        const gamesFromServerManifest:GameDescriptor[] = await getGameManifest(); 
-        gamesFromServerManifest.forEach(g => {
-            g.lazyType = React.lazy(() => import( (g as any).codeUrl))
+        const gamesFromServerManifest:GameManifestItem[] = await getGameManifest(); 
+
+        const gameList: GameDescriptor[] = []
+        gamesFromServerManifest.forEach(serverItem => {
+            const foundGame = allGames.find(g => g.name.toLowerCase() === serverItem.name.toLowerCase()) 
+            if(foundGame) {
+                const addMe = {...foundGame}
+                if(serverItem.displayName) addMe.displayName = serverItem.displayName
+                addMe.tags = serverItem.tags;
+                gameList.push(addMe);
+            }
+            else {
+                console.log(`Server specified a game I don't know about: ${serverItem.name}`)
+            }
```

Comment:
Nit: This can be a map/filter:

    const gameList = gamesFromServerManifest
        .map(serverItem => { /* return addMe if foundGame, return null otherwise */ })
        .filter(gameDescriptor => gameDescriptor);